### PR TITLE
F/tests

### PR DESCRIPTION
**Pull request by cloudformdesign**
Please merge this as soon as possible

### CODE REVIEW

File: pynio/service.py [hunk 1]
```diff
@@ -38,6 +38,13 @@ def connect(self, blk1, blk2):
             if blk['name'] == blk1.name:
                 connection = blk
                 break
+        for blk in execution:
+            if blk['name'] == blk2.name:
+                break
+        else:
+            # block2 doesn't exist, so add it
+            execution.append({'name': blk2.name, 'receivers': []})
```
Comment (hansmosh) [inline pynio/service.py]: I'm pretty sure this isn't necessary. Can you verify that nio actually requires the receiving block to be added to the service?

**Comment by cloudformdesign** [comment]
hold on a sec

File: pynio/service.py (hunk 1 above)
Comment (cloudformdesign) [inline pynio/service.py]: only the UI requires it. Might as well not break the UI

File: tests/test_block.py [hunk 2]
```diff
@@ -22,10 +22,12 @@ def test_save(self):
...
             b = Block('name', 'type')
             b.config = {'key': 'val'}
             b._put = MagicMock()
-            b.save('instance')
-            self.assertTrue(b._put.called)
-            self.assertEqual(b._instance, 'instance')
+            with self.assertRaises(Exception) as context:
+                b.save()
+            self.assertTrue('Block is not associated with an instance',
+                            context.exception)
```
Comment (cloudformdesign) [inline tests/test_block.py]: I think you have this backwards. Right now you are asserting that a string is True

Comment (cloudformdesign) [inline tests/test_block.py]: I'm actually not sure what this line is doing in the first place. Why would you assert that you have the exception?

Comment (hansmosh) [inline tests/test_block.py]: Good catch. Fixed.

**Pull request merged**